service-04/PUS-4-2.asn1

```diff
@@ -0,0 +1,61 @@
+--!
+--! Copyright (C) 2017 N7 Mobile sp. z o. o.
+--! Contact: http://n7mobile.pl/Space
+--!
+--! This file is part of ASN.1/ACN PUS-C Components Library.
+--!
+--! Library was developed under a programme and funded by
+--! European Space Agency.
+--!
+--! This Library is free software: you can redistribute it and/or modify
+--! it under the terms of the GNU General Public License as published by
+--! the Free Software Foundation, either version 3 of the License, or
+--! (at your option) any later version.
+--!
+--! This Library is distributed in the hope that it will be useful,
+--! but WITHOUT ANY WARRANTY; without even the implied warranty of
+--! MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+--! GNU General Public License for more details.
+--!
+--! You should have received a copy of the GNU General Public License
+--! along with this program.  If not, see <http://www.gnu.org/licenses/>.
+--!
+
+PUS-4-2 DEFINITIONS AUTOMATIC TAGS ::= BEGIN
+EXPORTS ALL;
+IMPORTS
+    AbsoluteTime, PUSC-UINT32 FROM BasicTypes
+    Parameter-ID, ParameterValue FROM Parameter;
+
+maxTC-4-2-StatisticsCount INTEGER ::= 2147483647
+
+TC-4-2-ParameterStatisticsReport ::= TC-4-2-ParameterStatisticsReportGeneric {AbsoluteTime, Statistics}
+
+Statistics ::= StatisticsGeneric {Parameter-ID, PUSC-UINT32, ParameterValue, AbsoluteTime, StandardDeviation}
+
+TC-4-2-ParameterStatisticsReportGeneric {TAbsoluteTime, TStatistics} ::= SEQUENCE
+{
+    startTime TAbsoluteTime,
+    endTime TAbsoluteTime,
+--! n UnsignedInteger, moved to acn
+    statistics SEQUENCE (SIZE(0 .. maxTC-4-2-StatisticsCount)) OF TStatistics
+}
+
+StatisticsGeneric {TParameter-ID, UnsignedInteger, TParameterValue, TAbsoluteTime, TStandardDeviation} ::= SEQUENCE
```

Comment:
are those `T` prefixes necessary?

Reply:
Should I change it to something else? I'm importing from other modules types like Parameter-ID, so in that case names will collide if I remove `T` prefix

Reply:
oh, yeah... it's still in the same module so there are possible conflicts.... I'm just not very fond of any superfluous suffixes or prefixes that come with no additional information and obscure possible "saying it out loud" ... damn... no idea how to solve it properly

Reply:
What about all upper case letters and `-`? Asn.1 constants are named with other convention anyway, since they must start with lower case.

Reply:
all upper case will be harder to read imho, so maybe something like `Parameter-ID-Type`? If it won't work - stick with `T`